src/AppBundle/Service/Mailer/MailSenderMock.php

```diff
@@ -3,44 +3,70 @@
 namespace AppBundle\Service\Mailer;
 
 use AppBundle\Model\Email;
+use Predis\Client as PredisClient;
 use Swift_Attachment;
+use Swift_Mailer;
 use Swift_Message;
-use Symfony\Bundle\FrameworkBundle\Translation\Translator;
-use Symfony\Component\DependencyInjection\Container;
-use Symfony\Component\Validator\Validator;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
 
 class MailSenderMock extends MailSender
 {
-    private static $messagesSent = [];
-
     /**
-     * @var Translator
+     * REDIS key used to store email mocks
      */
-    protected $translator;
+    const REDIS_EMAIL_KEY = 'behatEmailMock';
 
     /**
-     * @var Container
+     * @var Swift_Mailer[]
      */
-    protected $container;
+    private $mailers = [];
 
     /**
-     * @var \Symfony\Bundle\FrameworkBundle\Routing\Router
+     * @var PredisClient
      */
-    protected $router;
+    private $redis;
 
+    /**
+     * @var Symfony\Component\Validator\Validator
+     */
     protected $validator;
 
     /**
+     * @param ValidatorInterface $validator
      * @param \AppBundle\Mailer\MailerService $apiClient
-     * @param Translator                      $translator
      */
-    public function __construct(Container $container)
+    public function __construct(ValidatorInterface $validator, PredisClient $redis)
+    {
+        $this->mailers = [];
+        $this->validator = $validator;
+        $this->redis = $redis;
+    }
+
+    /**
+     * @param string       $name
+     * @param Swift_Mailer $mailer
+     */
+    public function addSwiftMailer($name, Swift_Mailer $mailer)
```

Comment:
I believe this same method is already in the parent. That said, if we move to an interface, this probably ought to be a method on that interface. The implementation needs a little further thought; naming it add `addSwiftMailer` and restricting it by typehint doesn't actually let us add any other mailers, which I presume is the point of the `->mailers` property. If we want this option to add more mailers I think it should be along the lines of `public function addMailer($name, MailerInterface $mailer)`, and we create a `SwiftMailer` which implements the `MailerInterface` and wraps the `Swift_Mailer` class.

Reply:
Have moved this suggestion to a new ticket ([DDPB-2630](https://opgtransform.atlassian.net/browse/DDPB-2630)) to avoid scope creep on this story and because the current implementation, whilst not perfect, is functional.